File: Roman Numeral Parser.java
import java.util.*;

public class RomanParser {
    public static int parseRomanNumeral(String s) {
        Map<Character, Integer> values = new HashMap<>();
        values.put('I', 1);
        values.put('V', 5);
        values.put('X', 10);
        values.put('L', 50);
        values.put('C', 100);
        values.put('D', 500);
        values.put('M', 1000);

        int total = 0;
        int i = 0;

        while (i < s.length()) {
            // Check if this char is smaller than the next one → subtract
            if (i + 1 < s.length() && values.get(s.charAt(i)) < values.get(s.charAt(i + 1))) {
                total += values.get(s.charAt(i + 1)) - values.get(s.charAt(i));
                i += 2;
            } else {
                total += values.get(s.charAt(i));
                i++;
            }
        }

        return total;
    }

   
}

File: Roman Numeral- Parser.c
#include <stdio.h>
#include <string.h>

int value(char c) {
    switch (c) {
        case 'I': return 1;
        case 'V': return 5;
        case 'X': return 10;
        case 'L': return 50;
        case 'C': return 100;
        case 'D': return 500;
        case 'M': return 1000;
        default: return 0;
    }
}

int parseRomanNumeral(const char *s) {
    int total = 0;
    int len = strlen(s);
    for (int i = 0; i < len; i++) {
        int curr = value(s[i]);
        int next = (i + 1 < len) ? value(s[i + 1]) : 0;

        if (curr < next) {
            total += next - curr;
            i++; // skip the next one
        } else {
            total += curr;
        }
    }
    return total;
}

